test/mod/test_util_chroot.py

```diff
@@ -0,0 +1,34 @@
+#
+# Test for util/chroot.py
+#
+
+from unittest.mock import patch
+
+from osbuild.util.chroot import Chroot
+
+
+class RunReturn:
+    """
+    Class to be returned from mocked run() call so that the returncode is always 0.
+    """
+
+    @property
+    def returncode(self):
+        return 0
+
+
+@patch("subprocess.check_call")
+@patch("subprocess.run", return_value=RunReturn())
+def test_chroot_context(mocked_run, mocked_check_call, tmp_path):
+
+    with Chroot(str(tmp_path)) as chroot:
+        chroot.run(["/bin/true"])
+
+    assert mocked_run.call_count == 4  # the chroot.run() call + 3 umount calls
```

Comment:
(super nitpick) probably a bit of matter of personal preferences, but how do you think this would look as a fully explicit test? i.e. just comparing `mocked_run.call_args_list in full`:
```python
#
# Test for util/chroot.py
#
import os

from unittest.mock import call, Mock, patch

from osbuild.util.chroot import Chroot


@patch("subprocess.run")
def test_chroot_context(mocked_run, tmp_path):
    mocked_run_result = Mock()
    mocked_run_result.returncode = 0
    mocked_run.return_value = mocked_run_result

    with Chroot(str(tmp_path)) as chroot:
        chroot.run(["/bin/true"], check=False, stdin="is-passed")

    assert mocked_run.call_args_list == [
        call(["/usr/bin/mount",
              "-t", "proc",
              "-o", "nosuid,noexec,nodev",
              "proc", os.fspath(tmp_path / "proc")], check=True),
        call(["/usr/bin/mount",
              "-t", "devtmpfs",
              "-o", "mode=0755,noexec,nosuid,strictatime",
              "devtmpfs", os.fspath(tmp_path / "dev")], check=True),
        call(["/usr/bin/mount",
              "-t", "sysfs",
              "-o", "nosuid,noexec,nodev",
              "sysfs", os.fspath(tmp_path / "sys")], check=True),

        call(["/usr/sbin/chroot", os.fspath(tmp_path), "/bin/true"], check=False, stdin="is-passed"),

        call(["/usr/bin/umount", "--lazy", os.fspath(tmp_path / "proc")], check=False),
        call(["/usr/bin/umount", "--lazy", os.fspath(tmp_path / "dev")], check=False),
        call(["/usr/bin/umount", "--lazy", os.fspath(tmp_path / "sys")], check=False),
    ]
```
(which of course needs a small tweak in chroot.py:
```python
diff --git a/osbuild/util/chroot.py b/osbuild/util/chroot.py
index d2033a29..20eea41d 100644
--- a/osbuild/util/chroot.py
+++ b/osbuild/util/chroot.py
@@ -21,18 +21,18 @@ class Chroot:
                 print(f"Making missing chroot directory: {d}")
                 os.makedirs(self.root + d)
 
-        subprocess.check_call(["/usr/bin/mount",
-                               "-t", "proc",
-                               "-o", "nosuid,noexec,nodev",
-                               "proc", f"{self.root}/proc"])
-        subprocess.check_call(["/usr/bin/mount",
-                               "-t", "devtmpfs",
-                               "-o", "mode=0755,noexec,nosuid,strictatime",
-                               "devtmpfs", f"{self.root}/dev"])
-        subprocess.check_call(["/usr/bin/mount",
-                               "-t", "sysfs",
-                               "-o", "nosuid,noexec,nodev",
-                               "sysfs", f"{self.root}/sys"])
+        subprocess.run(["/usr/bin/mount",
+                        "-t", "proc",
+                        "-o", "nosuid,noexec,nodev",
+                        "proc", f"{self.root}/proc"], check=True)
+        subprocess.run(["/usr/bin/mount",
+                        "-t", "devtmpfs",
+                        "-o", "mode=0755,noexec,nosuid,strictatime",
+                        "devtmpfs", f"{self.root}/dev"], check=True)
+        subprocess.run(["/usr/bin/mount",
+                        "-t", "sysfs",
+                        "-o", "nosuid,noexec,nodev",
+                        "sysfs", f"{self.root}/sys"], check=True)
 
         return self
```)

Reply:
Switching all subprocess methods to `subprocess.call()` is probably a good idea for consistency.  I don't know about checking every call arg list explicitly.  At that point it just looks like the test is 90% a copy of the code it's testing and it just seems silly.  And while I don't expect the mount call options to change frequently (or at all), it does make me question the utility of needing to change the same detail in two places (the `Chroot` class and its test) whenever we do.

Reply:
Thanks! I leave that to you how exactly it should look. My reasoning (sorry, I may have not made this very clear) is mostly that (to me) the explicit call_arg list is very simple to read, i.e. I don't need to think about the loop and how exactly it works. To me the fact that the test needs to change if the code changes and now e.g. changes a mount option is more a feature than an issue (as long as it only needs changing in a single place) but I understand that people have different philosophies here :)

Reply:
Ok ok, you changed my mind :)

Reply:
I hope you did not feel pressured or anything! Thanks for the change though :)